genai-perf/genai_perf/inputs/converters/openai_chat_completions_converter.py

```diff
@@ -30,55 +30,66 @@
 from genai_perf.inputs.converters.base_converter import BaseConverter
 from genai_perf.inputs.input_constants import DEFAULT_OUTPUT_TOKENS_MEAN, OutputFormat
 from genai_perf.inputs.inputs_config import InputsConfig
-
+from genai_perf.inputs.retrievers.generic_dataset import DataRow, GenericDataset
+from genai_perf.exceptions import GenAIPerfException
+from genai_perf.inputs.input_constants import DEFAULT_BATCH_SIZE
 
 class OpenAIChatCompletionsConverter(BaseConverter):
 
-    def convert(self, generic_dataset: Dict, config: InputsConfig) -> Dict:
+    def check_config(self, config: InputsConfig) -> None:
+        if config.output_format == OutputFormat.IMAGE_RETRIEVAL:
+            if config.add_stream:
+                raise GenAIPerfException(f"The --streaming option is not supported for {config.output_format.to_lowercase()}.")
+        else:
+            if config.batch_size_text != DEFAULT_BATCH_SIZE:
+                raise GenAIPerfException(f"The --batch-size-text flag is not supported for {config.output_format.to_lowercase()}.")
+            if config.batch_size_image != DEFAULT_BATCH_SIZE:
+                raise GenAIPerfException(f"The --batch-size-image flag is not supported for {config.output_format.to_lowercase()}.")
+
+
+    def convert(self, generic_dataset: GenericDataset, config: InputsConfig) -> Dict[Any, Any]:
         request_body: Dict[str, Any] = {"data": []}
 
-        for index, entry in enumerate(generic_dataset["rows"]):
-            model_name = self._select_model_name(config, index)
+        for file_data in generic_dataset.files_data.values():
+            for index, row in enumerate(file_data.rows):
+                model_name = self._select_model_name(config, index)
 
-            content: Any = []
-            if config.output_format == OutputFormat.OPENAI_CHAT_COMPLETIONS:
-                content = entry["text"]
-            else:
-                # Treat single batch and multi-batch entries the same way
-                entries = entry if isinstance(entry, list) else [entry]
-                for _entry in entries:
-                    content += self._add_multi_modal_content(_entry)
+                content: Any = []
+                if config.output_format == OutputFormat.OPENAI_CHAT_COMPLETIONS:
+                    content = row.texts[0]
+                else:
+                    content += self._add_multi_modal_content(row)
 
-            payload = {
-                "model": model_name,
-                "messages": [
-                    {
-                        "role": "user",
-                        "content": content,
-                    }
-                ],
-            }
+                payload = {
```

Comment:
Done. I only did it for this converter. All convert is doing is creating the payload, so it's up to the individual converters to decide how to do so. In this case, since there was still more complexity and lines of code due to this converter having multiple output formats, I pulled the logic out.